### PR TITLE
Rust: add the option to install additional targets

### LINKED ISSUE #1032: Rust: add the option to install additional targets
It'd be handy if there was a `target` option in the [Rust](https://github.com/devcontainers/features/tree/main/src/rust) package for when I want to add additional targets.

For each element it'd run `rustup target add ${target}`.

### PR DESCRIPTION
**Pull request by yaleman**
Fixes #1032

* Adds the "targets" input
* Did some shellcheck-based linting of the shell script
* Added a test for the new option

### CODE REVIEW

**Comment by yaleman** [comment]
@microsoft-github-policy-service agree

File: src/rust/devcontainer-feature.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
     "id": "rust",
-    "version": "1.1.3",
+    "version": "1.1.4",
```
Comment (samruddhikhandale) [inline src/rust/devcontainer-feature.json:3]:
```suggestion
    "version": "1.2.0",
```

Can we bump minor version as we are adding a new Feature option?

File: src/rust/devcontainer-feature.json [hunk 2]
```diff
@@ -39,6 +39,16 @@
             ],
             "default": "minimal",
             "description": "Select a rustup install profile."
+        },
+        "targets": {
+            "type": "string",
+            "default": "",
+            "description": "Comma-delimited list of additional Rust targets to install.",
```
Comment (samruddhikhandale) [inline src/rust/devcontainer-feature.json:46]:
```suggestion
            "description": "Optional comma separated list of additional Rust target to install",
```

File: src/rust/install.sh [hunk 3]
```diff
@@ -202,6 +205,13 @@ fi
 echo "Installing common Rust dependencies..."
 rustup component add rls rust-analysis rust-src rustfmt clippy 2>&1
 
+if [ -n "${RUSTUP_TARGETS}" ]; then
+    for target in $(echo "${RUSTUP_TARGETS}" | sed -E 's/,/ /g'); do
+        echo "Installing additional Rust target $target"
+        rustup target add "$target" 2>&1
+    done
+fi
```
Comment (samruddhikhandale) [inline src/rust/install.sh:213]:
```suggestion
if [ -n "${RUSTUP_TARGETS}" ]; then
    IFS=',' read -ra targets <<< "${RUSTUP_TARGETS}"
    for target in "${targets[@]}"; do
        echo "Installing additional Rust target $target"
        rustup target add "$target" 2>&1
    done
fi
```

Nit: Suggestion (untested)

**Review by samruddhikhandale — CHANGES_REQUESTED** [review]
Thanks for taking the time in contributing this PR. The new Feature option addition makes sense to me and will be very helpful.

Left some minor comments.
Also, thanks for optimizing the current script 🙏

**Review by samruddhikhandale — APPROVED** [review]
Wonderful, thank you ✨

**Pull request merged**